mev_inspect/classifiers/specs/zero_ex.py

```diff
@@ -165,3 +222,57 @@
 ]
 
 ZEROX_CLASSIFIER_SPECS = ZEROX_CONTRACT_SPECS + ZEROX_GENERIC_SPECS
+
+
+def _get_taker_token_in_amount(
+    taker_address: str, token_in_address: str, child_transfers: List[Transfer]
+) -> int:
+
+    if len(child_transfers) != 2:
+        raise ValueError(
```

Comment:
Even better than an exception is an interface that can _only be called correctly_

ex:
```
def _get_taker_token_in_amount(
    taker_address: str, token_in_address: str, first_transfer: Transfer, second_transfer: Transfer,
) -> int:
    ...
```

Reply:
To get the first and second transfer, i'd still need to check if the length of child_transfers to not get an IndexError, so why not include the exception in the function rather than out of it

Reply:
I think better to include the check outside the function and use the signature to say "here's what I need"

Reply:
I disagree, I think keeping the exception inside the function makes more sense. There was an error getting the token_in_data because child_transfers is too long. Keeps it more specific and contextualizes the exception with the use of child_transfers directly below. Makes the necessity for this assumption and its impact clearer in the code

Reply:
Here's better description of why it feels bad to me

If I already have two transfers `first` and `second` this function interface requires that I call the function like
```
_get_taker_token_in_amount(..., [first, second])
```
basically merging it together, to then check a condition which was previously definitely met and now is only maybe met, then breaking it up again

having the exception inside is sort of a "surprise" constraint (i have to read the code to know what's valid) v. making it so i can only call it in a valid way.

Reply:
you don't know if you have first and second, until you check child_transfers, which is passed in as an argument. In reality you have:
```
_get_taker_token_in_amount(..., [?])
```

Reply:
why use arguments at all though v. `_get_taker_token_in_amount(args: List[Any])`?

it's the caller's fault that they happen to have a list

if `parse_swaps` was given a dictionary would you change this function to take a dictionary?

if there were two callers where one has a list and one has a dictionary what would be the interface then?

the interface should describe what the function needs, and it shouldn't be possible to call a function with invalid parameters

otherwise i don't know i've passed bad values till a runtime error - instead i could just have that bug not be possible

Reply:
it's not. The get_child_traces returns a lit in swaps.py. That list is used for this. But for brevity I rather us close this issue and settle on the solution so we can ship 0x. I don't feel too strongly about this